### DISCUSSION TITLE
Customize html templates and remove expiration controls

### DISCUSSION THREAD

**Discussion by vibarco** [pglombardo/PasswordPusher#1698] [Ideas]
Hello,

is it possible to modify #404 html templates? where are templates located?. I am also interested in the possibility to remove the controls from the otp URL generator backend. I would like to avoid the user to set expiration parameters (attached).

Best
![remove_controls](https://github.com/pglombardo/PasswordPusher/assets/33134255/aa0cf931-ac49-4798-986c-084d445d3271)

**Comment by pglombardo** [comment]
Hi @vibarco,

Re: 404 pages, they are located [here](https://github.com/pglombardo/PasswordPusher/blob/master/public).  You can overwrite those files in your container if needed.  The path in the container is `/opt/PasswordPusher/public`.

>  I am also interested in the possibility to remove the controls from the otp URL generator backend. I would like to avoid the user to set expiration parameters (attached).

This isn't possible currently.  As an alternative, you can limit the range of those values to say "only between 2-3 days" and "1-3" views using the settings in [settings.yml](https://github.com/pglombardo/PasswordPusher/blob/master/config/settings.yml#L206-L208).

**Reply by vibarco** [reply]
Many Thanks Pete,

much appreciated.

Best
